// world/config/client/config.js
'use strict';

module.exports = {
    events: {
        dom: {
            world: {
                event: 'ready',
                selector: '#world',
                sequences: [
                    {
                        name: 'display',
                        input: {
                            content: 'World'
                        }
                    }
                ]
            },
            exclamation: {
                event: 'ready',
                selector: '#world-exclamation',
                sequences: [
                    {
                        name: 'display',
                        input: {
                            content: require('exclamation/core/content')
                        }
                    }
                ]
            }
        }
    },
    sequences: {
        display: {
            stream: {
                content: {
                    type: 'string',
                    required: true
                }
            },
            operations: [
                {
                    order: 0,
                    service: 'danf:manipulation.callbackExecutor',
                    method: 'execute',
                    arguments: [
                        function(target, content) {
                            target.innerText = content;
                            target.removeAttribute('style');
                        },
                        '!event.target!',
                        '@content@'
                    ]
                }
            ]
        }
    }
};
